app/experimenter/experiments/tests/test_serializers.py

```diff
@@ -1433,17 +1433,16 @@ def test_serializer_rejects_duplicate_branch_names(self):
         self.assertFalse(serializer.is_valid())
         self.assertIn("variants", serializer.errors)
 
-    def test_serializer_rejects_special_char_branch_names(self):
+    def test_serializer_allows_special_char_branch_names(self):
         experiment = ExperimentFactory.create(type=ExperimentConstants.TYPE_PREF)
 
-        self.control_variant_data["name"] = "&re@t br@nche$!"
+        self.control_variant_data["name"] = "&re@t -br@nche$!"
 
         data = {"variants": [self.control_variant_data, self.treatment_variant_data]}
 
         serializer = ExperimentDesignBaseSerializer(instance=experiment, data=data)
 
-        self.assertFalse(serializer.is_valid())
-        self.assertIn("variants", serializer.errors)
+        self.assertTrue(serializer.is_valid())
```

Comment:
I would also save and check that the 

- name is preserved
- the slug is the expected value